### PR TITLE
substantial changes to wikipedia fetching code

### PR DESCRIPTION
**Pull request by makoshark**
- Added new code to fetch revisions data from Wikipedia.
- Refactored the view code to (a) make it write/process things as a stream and (b) write data as a stream of JSON dictionaries instead of a single JSON list.
- Renamed files to make it clear it's currently just English Wikipedia.
- Many other small changes.

This definitely breaks the example code for page views which will need to get fixed.

### CODE REVIEW

File: wikipedia/scripts/fetch_enwiki_daily_views.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+###############################################################################
+
+import sys
+import subprocess
+import requests
+import argparse
+import json
+import time
+import os.path
+import argparse
```
Comment (groceryheist) [inline wikipedia/scripts/fetch_enwiki_daily_views.py]: importing argparse twice

File: wikipedia/scripts/fetch_enwiki_revisions.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    #handle -L
+    loglevel_mapping = { 'debug' : logging.DEBUG,
+                         'info' : logging.INFO,
+                         'warning' : logging.WARNING,
+                         'error' : logging.ERROR,
+                         'critical' : logging.CRITICAL }
+
+    if args.logging_level in loglevel_mapping:
+        loglevel = loglevel_mapping[args.logging_level]
+    else:
```
Comment (groceryheist) [inline wikipedia/scripts/fetch_enwiki_revisions.py]: suggest moving this to a function used in both scripts.

File: wikipedia/scripts/fetch_enwiki_revisions.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                 'contentmodel' : 'contentmodel',
+                 'tags' : 'tags',
+                 'comment' : 'comment',
+                 'content' : 'content' }
+
+    exclude_from_tsv = ['tags', 'comment', 'content']
+
+    # load the list of articles
+    with open(article_filename, 'r') as infile:
+        article_list = [art.strip() for art in list(infile)]
```
Comment (groceryheist) [inline wikipedia/scripts/fetch_enwiki_revisions.py]: This does two passes over infile. First in the call to list, then in the listcomp

i would do:
list(map(str.strip, infile))

File: wikipedia/scripts/wikiproject_scraper.py [hunk 4]
```diff
@@ -28,48 +29,45 @@
...
+        print("Choose a valid log level: debug, info, warning, error, or critical") 
         exit
+        
+    #handle -W
+    if args.logging_destination:
+        logging.basicConfig(filename=args.logging_destination, filemode='a', level=loglevel)
+    else:
+        logging.basicConfig(level=loglevel)
 
+    export_git_hash = subprocess.check_output(['git', 'rev-parse', 'HEAD']).decode().strip()
```
Comment (groceryheist) [inline wikipedia/scripts/wikiproject_scraper.py:51]: also suggest using a shared function for git hashes.

**Review by groceryheist — APPROVED** [review]

**Comment by makoshark** [comment]
I've made all the requested changes and added several new things including:

- bash scripts to run these files in cronjobs
- a bunch of smaller bug fixes

**Comment by makoshark** [comment]
@groceryheist You might want to take another look and then merge these over if you think it's ready.

File: wikipedia/scripts/digobs.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import subprocess
+import logging
+
+def git_hash(short=False):
+    if short:
+        return subprocess.check_output(['git', 'rev-parse', '--short', 'HEAD']).decode().strip()
+    else:
+        subprocess.check_output(['git', 'rev-parse', 'HEAD']).decode().strip()
+
+def get_loglevel(arg_loglevel):
```
Comment (groceryheist) [inline wikipedia/scripts/digobs.py:13]: So it turns out that argparse has a cool feature where you can pass in a function that converts a string to an object and get that in your args object.  So, in theory we could pass in type=get_loglevel and it should put the loglevel in the args.

File: wikipedia/scripts/fetch_enwiki_daily_views.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+from csv import DictWriter
+import digobs
+#import feather #TBD
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Call the views API to collect Wikipedia view data.')
+    parser.add_argument('-o', '--output_folder', help='Where to save output', default="wikipedia/data", type=str)
+    parser.add_argument('-i', '--article_file', help='File listing article names', default="wikipedia/resources/enwp_wikiproject_covid19_articles.txt", type=str)
+    parser.add_argument('-d', '--query_date', help='Date if not yesterday, in YYYYMMDD format.', type=str)
+    parser.add_argument('-L', '--logging_level', help='Logging level. Options are debug, info, warning, error, critical. Default: info.', default='info', type=str),
```
Comment (groceryheist) [inline wikipedia/scripts/fetch_enwiki_daily_views.py]: see above comment about how the type= param can be useful here.

File: wikipedia/scripts/wikiproject_scraper.py [hunk 7]
```diff
@@ -28,48 +29,45 @@
...
-        logging.basicConfig(filename=logHome, filemode='a', level=logging.DEBUG)
-    elif loglevel == 'info':
-        logging.basicConfig(filename=logHome, filemode='a', level=logging.INFO)
-    elif loglevel == 'warning':
-        logging.basicConfig(filename=logHome, filemode='a', level=logging.WARNING)
-    elif loglevel == 'error':
-        logging.basicConfig(filename=logHome, filemode='a', level=logging.ERROR)
-    elif loglevel == 'critical':
-        logging.basicConfig(filename=logHome, filemode='a', level=logging.CRITICAL)
+    loglevel_mapping = { 'debug' : logging.DEBUG,
```
Comment (groceryheist) [inline wikipedia/scripts/wikiproject_scraper.py]: We should reuse digobs.get_loglevel here.

File: wikipedia/scripts/digobs.py (hunk 5 above)
Comment (makoshark) [reply wikipedia/scripts/digobs.py:13]: OK. I've done this.

**Pull request merged**